providers/Flipt/src/FliptProvider.php

```diff
@@ -0,0 +1,137 @@
+    {
+        return $this->resolveValue($flagKey, FlagValueType::OBJECT, $defaultValue, $context);
+    }
+
+
+    /**
+     * Clears the cache of all requests
+     */
+    public function clearCache() {
+        $this->cache->clear();
+    }
+
+    /**
+     * @param bool|string|int|float|mixed[] $defaultValue
+     */
+    private function resolveValue(string $flagKey, string $flagType, mixed $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
+    {
+
+        // check null context
+        if( empty( $context ) ) {
+            $attributes = [];
+            $id = null;
+        } else {
+            $attributes = $context->getAttributes()->toArray();
+            $id = $context->getTargetingKey();
+        }
+
+
+        // check if cache has already result
+        $cacheKey = $this->cache->key( [ 'flag' => $flagKey, 'type' => $flagType, 'default' => $defaultValue, 'context' => $attributes, 'key' => $id ] );
+        $cached = $this->cache->get( $cacheKey );
+        if( isset( $cached ) ) return $cached;
+
+
+        // booleans need a dedicated function
+        if( $flagType == FlagValueType::BOOLEAN ) {
+            $result = $this->client->boolean( $flagKey, $attributes, $id );
+        } else {
+            $result = $this->client->variant( $flagKey, $attributes, $id );
+        }
+
+        
+        // there is a match
+        // not sure yet as the variant result has a getMatch() but not the boolean result.
+        if( $result->getReason() == 'MATCH_EVALUATION_REASON' || $result->getReason() == "DEFAULT_EVALUATION_REASON" ) {
+            $result = ResolutionDetailsFactory::fromSuccess( $this->castResult( $result, $flagType ) );
+        } else {
+            $result = (new ResolutionDetailsBuilder())
+                    ->withValue( $defaultValue )
+                    ->withError(
+                        // not sure if thie reason to error mapping is correct
+                        new ResolutionError(ErrorCode::GENERAL(), $result->getReason() ),
+                    )
+                    ->build();
+        }
+
+        // write result into cache
+        $this->cache->set( $cacheKey, $result ); 
+
+        return $result;
+    }
+
+
+
+    private function castResult( VariantEvaluationResult|BooleanEvaluationResult $result, string $type ) {
+        switch ($type) {
+            case FlagValueType::BOOLEAN:
+                return filter_var($result->getEnabled(), FILTER_VALIDATE_BOOLEAN);
+            case FlagValueType::FLOAT:
+                return (float) $result->getVariantKey();
+            case FlagValueType::INTEGER:
+                return (int) $result->getVariantKey();
+            case FlagValueType::OBJECT:
+                return json_decode( $result->getVariantAttachment(), true);
+            case FlagValueType::STRING:
+                return $result->getVariantKey();
+            default:
```

Comment:
This should never happen, in which case if it did, it should be an error not nullng the value.

Reply:
Im not sure if this is an issue as I use this only when I get a successfull response from the client:
```php
if( $result->getReason() == ResponseReasons::MATCH_EVALUATION_REASON || $result->getReason() == ResponseReasons::DEFAULT_EVALUATION_REASON ) {
            $result = ResolutionDetailsFactory::fromSuccess( $this->castResult( $result, $flagType ) );
}
```

And the I suppose a correct value. But you are in a way right if the response value is somewhat invalid I should throw an error. What do you propose here?